web/src/message_list.js

```diff
@@ -72,6 +69,10 @@ export class MessageList {
         // the user. Possibly this can be unified in some nice way.
         this.reading_prevented = false;
 
+        // Used to determine whether to preserve the rendered state
+        // when this message list is not the current message list.
+        this.preserve_rendered_state = user_settings.web_home_view === "all_messages" && !this.narrowed;
```

Comment:
Probably worth extending the comment to mention that this doesn't get live-updated when the setting changes, but that's probably OK, as it's just an optimization?

Reply:
Oh! I forgot to add a comment here. I was thinking if user had `all_messages` as the default view in the current session, then it doesn't make sense for us to no longer have it's rendered state be present if they changed the default view. Since, user likely still uses the combined feed view and the faster rendering of combined feed view could be a better experience if we have already done the work of keeping it updated.

Reply:
I can update `preserve_rendered_state` in event handlers code if you think otherwise.

Reply:
Yeah I think the current logic is good, we should just make sure the subtlety is documented.